### DISCUSSION TITLE
share script : MyriadConnect update automation

### DISCUSSION THREAD

**Discussion by info100t** [amidaware/tacticalrmm#1129] [General]
hi everybody,

First... sorry for my poor english... :(

Because i'm not a software dev, i'll try to share scripts automation i'll create.

So the first...

For our customer, we don't sell RDS ou Citrix licences. We use and ressels : https://myriad-connect.com/

My automation ps script :

```
## creation du dossier temp
$Folder = 'C:\SCRIPTS\TEMP'
if (Test-Path -Path $Folder) {
Write-Host("Le dossier temporaire existe") -fore green
} else {
Write-Host("Creation du dossier temporaire") -fore red
mkdir C:\SCRIPTS\TEMP
}

## on recupere la derniere version disponible en téléchargeant un fichier txt chez l'éditeur
Write-Host("Telechargement du fichier referentiel de mise a jour publiee") -fore green
wget "https://dl-files.com/latestversion.txt" -outfile "C:\SCRIPTS\TEMP\mc_verupdate.txt"

## on exporte la version installée sur le système
Write-Host("Generation du fichier de la mise a jour installee sur le systeme") -fore green
Get-ItemPropertyValue -Path "HKLM:\SOFTWARE\Digital River\JWTS" -Name Version > C:\SCRIPTS\TEMP\mc_versysteme.txt

## on compare les versions
Write-Host("Comparaison des versions") -fore green
$valeur1 = Get-Content "C:\SCRIPTS\TEMP\mc_versysteme.txt"
$valeur2 = Get-Content "C:\SCRIPTS\TEMP\mc_verupdate.txt"
if ($valeur1.ToString() -ieq $Valeur2.ToString()) 
{
Write-Host("Derniere version installee sur le systeme") -fore green
Remove-Item C:\SCRIPTS\TEMP\*.*
exit
}
else{
Write-Host("Installation de la derniere version sur le systeme et redemarrage") -fore red
wget https://dl-files.com/data/UpdateRelease.exe -outfile "C:\SCRIPTS\TEMP\UpdateRelease.exe"
c:\SCRIPTS\TEMP\UpdateRelease.exe /verysilent /Addon no
}
```

**Comment by silversword411** [comment]
Thanks for the share. It's going to be difficult as-is to do any integration. 

If you want to get this added to the docs maybe you can write something up similar to: https://docs.tacticalrmm.com/3rdparty_screenconnect/

If you're wanting to put a script into the script library about this, it'll probably need to be: 
- translated to english
- Don't recommend using `C:\SCRIPTS\TEMP` use `$env:programdata\TacticalRMM\temp\`

Then you'll want to do a PR to: https://github.com/amidaware/community-scripts

If you need help with this, recommend joining Discord and discuss in the scripts channel
